packages/frontend/lib/index.ts

```diff
@@ -202,25 +217,53 @@ export default class Nango {
                 { width: this.width, height: this.height },
                 this.debug
             );
-            if (options?.detectClosedAuthWindow || false) {
+
+            if (options?.detectClosedAuthWindow) {
                 this.tm = setInterval(() => {
-                    if (!this.win?.modal.window || this.win.modal.window.closed) {
-                        if (this.win?.isProcessingMessage === true) {
-                            // Modal is still processing a web socket message from the server
-                            // We ignore the window being closed for now
-                            return;
-                        }
-                        clearTimeout(this.tm as unknown as number);
-                        this.win = null;
-                        this.status = AuthorizationStatus.CANCELED;
-                        const error = new AuthError('The authorization window was closed before the authorization flow was completed', 'windowClosed');
-                        reject(error);
+                    if (!this.win || !this.win.modal) {
+                        return;
+                    }
+
+                    if (this.win.modal.window && !this.win.modal.window.closed) {
+                        return;
                     }
+
+                    if (this.win.isProcessingMessage) {
+                        // Modal is still processing a web socket message from the server
+                        // We ignore the window being closed for now
+                        return;
+                    }
+
+                    clearInterval(this.tm as unknown as number);
+                    this.win = null;
+                    this.status = AuthorizationStatus.CANCELED;
+                    const error = new AuthError('The authorization window was closed before the authorization flow was completed', 'windowClosed');
+                    reject(error);
                 }, 500);
             }
         });
     }
 
+    /**
+     * Clear state of the frontend SDK
+     */
+    public clear() {
+        if (this.tm) {
+            clearInterval(this.tm as unknown as number);
+        }
+
+        if (this.win) {
+            try {
+                this.win.close();
+            } catch {
+                // do nothing
+            }
+            this.win = null;
+        }
+
+        this.status = AuthorizationStatus.IDLE;
+    }
+
```

Comment:
I think that calling `clear` in `nango.auth()` will solve this bug: https://linear.app/nango/issue/NAN-1773/fix-the-authorization-window-is-already-opened